scipy/integrate/_quadrature.py

```diff
@@ -662,6 +664,216 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
     return result
 
 
+def _cumulative_simpson_equal_intervals(
+    y: np.ndarray, dx: np.ndarray, axis: int
+) -> np.ndarray:
+
+    y = y.swapaxes(0, axis)
+    dx = dx.swapaxes(0, axis)
+
+    sub_integrals_fwd = dx[:-1] / 3 * \
+        (5 * y[:-2] / 4 + 2 * y[1:-1] - y[2:] / 4)
+    sub_integrals_bwd = dx[:-1] / 3 * \
+        (-y[:-2] / 4 + 2 * y[1:-1] + 5 * y[2:] / 4)
```

Comment:
It would be nice to know where these formulae come from.
For instance, an explanation like: The first is the Simpson rule for the integration part from a to a+h, the second from a+h to a+2h=b (but note that wikipedia uses a different h, I think). Adding both parts gives the famous Simpson rule, see wikipedia....

Reply:
Thank you for the review @lorentzenchr !

Just to clarify, are you requesting I add this information as a comment above line 674? Or, can I add it elsewhere such as the docstring of the cumulative_simpson function?

Wikipedia uses the same 'h'. It is defined as the 'step size': (b-a)/2
![image](https://user-images.githubusercontent.com/7060774/227790626-26a1cc4e-cd20-40b7-a619-d92791d3e6af.png)

But, Wikipedia does not list the solution for unequal intervals. This is derived in this notebook: [Cumulative Simpson derivation](https://github.com/nprav/misc/blob/main/cumulative_simpson/Cumulative%20Simpson%20algorithm%20derivation.ipynb). Any suggestion for how I can reference this (if needed)?

Reply:
A scipy maintainer might want to step in.
My personal order of preference is:
1. A Notes section in the docstring where one can use LaTeX.
2. A comment in the code

For the formula with unequal steps, I don‘t know what‘s best. Is there a paper or book that can be referenced?

Reply:
Thanks @lorentzenchr 

My mistake, the solution actually IS available on [wikipedia](https://en.wikipedia.org/wiki/Simpson%27s_rule), but with slightly different notation. I can reference this.
![image](https://user-images.githubusercontent.com/7060774/227803039-3cafcc18-812b-4256-b379-30e0bf68b1d2.png)

The Notes section in the docstring with added references sounds good.
I can then add a single line comment to improve clarity in the code itself.

Does this work?

Reply:
Notes and additional line comments have been added to explain formulae and link to Wikipedia details.